internal/pkg/gateway/api.go

```diff
@@ -226,11 +227,25 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 //
 // If the transaction commit status cannot be returned, for example if the specified channel does not exist, a
 // FailedPrecondition error will be returned.
-func (gs *Server) CommitStatus(ctx context.Context, request *gp.CommitStatusRequest) (*gp.CommitStatusResponse, error) {
-	if request == nil {
+func (gs *Server) CommitStatus(ctx context.Context, signedRequest *gp.SignedCommitStatusRequest) (*gp.CommitStatusResponse, error) {
+	if signedRequest == nil {
 		return nil, status.Error(codes.InvalidArgument, "a commit status request is required")
 	}
 
+	request := &gp.CommitStatusRequest{}
+	if err := proto.Unmarshal(signedRequest.Request, request); err != nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid status request")
+	}
+
+	signedData := &protoutil.SignedData{
+		Data:      signedRequest.Request,
+		Identity:  request.Identity,
+		Signature: signedRequest.Signature,
+	}
+	if err := gs.policyChecker.CheckACL(resources.Event_FilteredBlock, request.ChannelId, signedData); err != nil {
+		return nil, status.Error(codes.FailedPrecondition, err.Error())
```

Comment:
- `codes.PermissionDenied` is a better fit here.
- Since you use `resources.Event_FilteredBlock` the underlying error message may be misleading to pass on as is, since that would perhaps say something like "no permission for event/FilteredBlock". Either you may like to define a new resource for this api or decorate the error message here (I personally like the former as it matches the current scheme of one-to-one mapping of api to acl entry).

Reply:
Sounds good to me. Where do I define resources types?

Reply:
You can look at this file https://github.com/hyperledger/fabric/blob/main/core/aclmgmt/resources/resources.go and follow the pattern for Event_FilteredBlock